mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/mosaic.kt

```diff
@@ -92,10 +95,35 @@ public suspend fun runMosaic(body: suspend MosaicScope.() -> Unit): Unit = corou
 		}
 	}
 
+	val terminal = Terminal()
+	val terminalInfo = mutableStateOf(
+		TerminalInfo(
+			size = TerminalInfo.Size(terminal.info.width, terminal.info.height)
+		)
+	)
+
+	launch(context = composeContext) {
+		while (true) {
+			if (terminal.info.updateTerminalSize()
+				&& (terminalInfo.value.size.width != terminal.info.width
+					|| terminalInfo.value.size.height != terminal.info.height)
```

Comment:
For `updateTerminalSize` documentation says that it returns `true` if the size has been updated, but in reality it returns `true` even if the size remains the same, so I added checks to avoid creating objects once again

Reply:
Should we file a bug on Mordant about that behavior then?

Reply:
In the implementation, this is how it is intended, but the documentation is a little unclear, so I created an issue https://github.com/ajalt/mordant/issues/135 with a proposal to correct it